fdbserver/workloads/ClogRemoteTlog.actor.cpp

```diff
@@ -0,0 +1,161 @@
+						auto dataLag = role["data_lag"].get_obj();
+						ASSERT(dataLag.contains("seconds"));
+						ASSERT(dataLag.contains("versions"));
+						TraceEvent("SSDataLag")
+						    .detail("Process", p.first)
+						    .detail("Role", role["role"].get_str())
+						    .detail("SecondLag", dataLag["seconds"].get_value<double>())
+						    .detail("VersionLag", dataLag["versions"].get_int64());
+						maxSSLagSec = std::max(maxSSLagSec, dataLag["seconds"].get_value<double>());
+					}
+				}
+			}
+		}
+		TraceEvent("MaxSSDataLag").detail("SecondLag", maxSSLagSec).detail("SecondThreshold", self->lagThresholdSec);
+		// TODO (praza): Uncomment this assert when gray failure detection is improved to automatically fix this issue
+		// ASSERT(maxSSLagSec < self->lagThresholdSec);
+		return Void();
+	}
+
+	ACTOR static Future<std::vector<IPAddress>> getRemoteSSIPs(Database db) {
+		state std::vector<IPAddress> ret;
+		Transaction tr(db);
+		tr.setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
+		tr.setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+		tr.setOption(FDBTransactionOptions::LOCK_AWARE);
+		std::vector<std::pair<StorageServerInterface, ProcessClass>> results =
+		    wait(NativeAPI::getServerListAndProcessClasses(&tr));
+		for (auto& [ssi, p] : results) {
+			if (ssi.locality.dcId().present() && ssi.locality.dcId().get() == g_simulator->remoteDcId) {
+				ret.push_back(ssi.address().ip);
+			}
+		}
+		return ret;
+	}
+
+	static std::vector<IPAddress> getRemoteTLogIPs(ClogRemoteTLog* self) {
+		std::vector<IPAddress> remoteTLogIPs;
+		for (const auto& tLogSet : self->dbInfo->get().logSystemConfig.tLogs) {
+			if (tLogSet.isLocal) {
+				continue;
+			}
+			for (const auto& tLog : tLogSet.tLogs) {
+				remoteTLogIPs.push_back(tLog.interf().address().ip);
+			}
+		}
+		return remoteTLogIPs;
+	}
+
+	ACTOR static Future<Void> clogRemoteTLog(ClogRemoteTLog* self, Database db) {
+		wait(delay(self->clogInitDelaySec));
+
+		// First, get all remote TLog IPs
+		state std::vector<IPAddress> remoteTLogIPs = getRemoteTLogIPs(self);
+		ASSERT(!remoteTLogIPs.empty());
+
+		// Then, get all remote SS IPs
+		std::vector<IPAddress> remoteSSIPs = wait(getRemoteSSIPs(db));
+		ASSERT(!remoteSSIPs.empty());
+
+		// Then, clog a random remote TLog -> SS network link
+		state IPAddress remoteTLogIP = remoteTLogIPs[deterministicRandom()->randomInt(0, remoteTLogIPs.size())];
+		state IPAddress remoteSSIP = remoteSSIPs[deterministicRandom()->randomInt(0, remoteSSIPs.size())];
+		TraceEvent("ClogRemoteTLog").detail("SrcIP", remoteTLogIP).detail("DstIP", remoteSSIP);
+		g_simulator->clogPair(remoteTLogIP, remoteSSIP, self->testDuration);
+
+		// After some time, unclog
+		wait(delay(self->clogDurationSec));
+		g_simulator->unclogPair(remoteTLogIP, remoteSSIP);
+		TraceEvent("UnclogRemoteTLog").detail("SrcIP", remoteTLogIP).detail("DstIP", remoteSSIP);
+
+		// We only clog once, so this actor never finishes
+		wait(Never());
```

Comment:
If this actor never finishes, then `workload()` can only exit because of a cancel due to timeout. Is this intentional?

Reply:
The cancel due to timeout (value is testDuration) is intentional. Overall intent is:

1. Run measurement and clog actors concurrently. 
2. If the measurement actor sees high SS lag, it asserts (will enable in feature PR). 
3. Clog actor's job is to just clog+unclog and then do nothing until end of test (based on timeout/testDuration value).

Any concerns with this approach?

Relatedly, I still want to ensure clog+unclog have already happened before end of test (based on timeout/testDuration). Theoretically, it looks like it could be possible that we are stuck somewhere and the test exits, but we haven't done clog+unclog. In this kind of situation, the test should fail. To fix this:

1. I can maintain explicit state that the clog+unclog happened, as well as we have enough measurement samples, and assert at test exit time that state value is as expected.
2. Try this approach: https://github.com/spraza/foundationdb/blob/392bad2bd31ba146e05641680929da2c10480bc5/fdbserver/workloads/ClogTlog.actor.cpp#L197-L201. But I am not sure how we pick `10` as in the workloadEnd calculation here: https://github.com/spraza/foundationdb/blob/392bad2bd31ba146e05641680929da2c10480bc5/fdbserver/workloads/ClogTlog.actor.cpp#L168. 

Any thoughts based on general practice of writing reliable simulation tests?